cmd/livepeer/starter/starter.go

```diff
@@ -1420,29 +1461,52 @@ func checkOrStoreChainID(dbh *common.DB, chainID *big.Int) error {
 	return nil
 }
 
-// Format of broadcasterPrices json
-// {"broadcasters":[{"ethaddress":"address1","priceperunit":1000,"pixelsperunit":1}, {"ethaddress":"address2","priceperunit":2000,"pixelsperunit":3}]}
-type BroadcasterPrices struct {
-	Prices []BroadcasterPrice `json:"broadcasters"`
-}
-
 type BroadcasterPrice struct {
-	EthAddress    string `json:"ethaddress"`
-	PricePerUnit  int64  `json:"priceperunit"`
-	PixelsPerUnit int64  `json:"pixelsperunit"`
+	EthAddress    string
+	PricePerUnit  *big.Rat
+	Currency      string
+	PixelsPerUnit int64
 }
 
 func getBroadcasterPrices(broadcasterPrices string) []BroadcasterPrice {
-	var pricesSet BroadcasterPrices
-	prices, _ := common.ReadFromFile(broadcasterPrices)
+	if broadcasterPrices == "" {
+		return nil
+	}
+
+	// Format of broadcasterPrices json
+	// {"broadcasters":[{"ethaddress":"address1","priceperunit":0.5,"currency":"USD","pixelsperunit":1}, {"ethaddress":"address2","priceperunit":0.3,"currency":"USD","pixelsperunit":3}]}
+	var pricesSet struct {
+		Broadcasters []struct {
+			EthAddress    string          `json:"ethaddress"`
+			PricePerUnit  json.RawMessage `json:"priceperunit"`
+			Currency      string          `json:"currency"`
+			PixelsPerUnit int64           `json:"pixelsperunit"`
+		} `json:"broadcasters"`
```

Comment:
1. The type is only used here, so I understand it's only a local file being used as a config.
2. It will actually not be parsed as a string, but still a number! I've only made it a `json.RawMessage` so we wouldn't have to parse it as a `float64` here and potentially lose precision over the "string" that was present in the JSON. The parsing with `new(big.Rat).SetString` would fail if this value is a JSON string, since it would contain surrounding `"`.
3. I've tested it on the unit tests only, but will do manually with some different configs.

Reply:
👍